shell/platform/android/io/flutter/embedding/android/FlutterImageView.java

```diff
@@ -0,0 +1,88 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.embedding.android;
+
+import android.content.Context;
+import android.graphics.Bitmap;
+import android.graphics.Canvas;
+import android.graphics.ColorSpace;
+import android.hardware.HardwareBuffer;
+import android.media.Image;
+import android.media.Image.Plane;
+import android.media.ImageReader;
+import android.os.Build;
+import android.view.View;
+import androidx.annotation.Nullable;
+import androidx.annotation.RequiresApi;
+
+class FlutterImageView extends View implements RenderSurface {
+  private final ImageReader imageReader;
+  @Nullable private Image nextImage;
+  private Image currentImage;
+
+  public FlutterImageView(Context context, ImageReader imageReader) {
+    super(context, null);
+    this.imageReader = imageReader;
+  }
+
+  @RequiresApi(api = Build.VERSION_CODES.KITKAT)
```

Comment:
nit: It looks like this entire class requires `KITKAT` due to the dependency on `ImageReader`.